articles/how-to-fix-and-prevent-sql-injection-in-wordpress/index.md

```diff
@@ -0,0 +1,194 @@
+
+#### What is SQLi?
+
+SQLi is the injection or insertion of SQL code usually via data transmitted from a website. When an SQLi attack is deployed successfully, it gives them access to sensitive information as well as ability to add/edit/delete data from a database. Generally, SQLi gives an attacker the possibility to execute all actions related to a database. SQLI takes advantage of PHP and SQL database hence posing a big risk to WordPress. Other than PHP, [the ASP language](https://www.w3schools.com/asp/default.ASP) is also prone to SQLi.
+
+An attacker makes use of data input by the user in to the database that isn't sufficiently masked and contains meta characters i.e. semicolons, quotation marks and double hyphens. These parameters have specific functions with which the SQL interpreter uses to execute external commands. SQLi targets old interfaces which often do not necessarily mask key parameters such as ASP and PHP which make them a perfect target for an SQLi attack.
+
+### Types of SQLi Attacks
+
+Now we know the basics of SQLi, let's dive into the various types of attacks and why they happen. There are several types of SQLi attacks and they are classified depending on their technique. They are;
+
+#### In-Band SQLi
+
+- Error-Based SQLi
+- Union Based Attack
+
+#### Inferential/Blind SQLi
+
+- Boolean Blind SQLi
+- Time-Based Attack
+
+#### Out-of-Band SQLi
+
+### Inferential/Blind SQLi
+
+In some scenarios the attacker will generate an error in an SQL query but the response may not be transmitted as he may expect to a webpage. The attacker will need to probe further into the error.
+
+In this type of SQLi, the attacker needs to send various queries to the database and see how it analyses data hence known as an Inferential attack or Blind SQLi. We shall look into two types of blind SQLi attacks namely: boolean SQLi and time-based SQLi.
+
+#### Boolean SQLi
+
+An attacker will send SQL request with an intent to list the database. If it results in an SQL error that has not been handled internally within the application, the webpage displays an error, loads partially or a blank page. In a boolean attack, the attacker will assess what part of a user's input are susceptible to SQLi through two versions of a boolean clause. They are;
+~~~~ sql
```

Comment:
Code snippet should be encased in three backticks ``` instead of tildes for proper Markdown code formatting. Please correct further code snippets as well.

Reply:
@jwk19, is there are particular reason to use four tildes instead of three backticks?

Reply:
@louisefindlay23 Yes I have used the tildes because they are specific for MySQL syntax for proper Markdown formatting. I have actually referred to a published article: [MySQL Query Performance Optimization Tips](https://www.section.io/engineering-education/mysql-query-performance-optimization-tips/)

Reply:
Thanks, @jwk19.